compiler/src/dmd/parse.d

```diff
@@ -7032,14 +7036,20 @@ class Parser(AST, Lexer = dmd.lexer.Lexer) : Lexer
                 }
                 break;
 
+            case TOK.endOfLine:
+                nextDefineLine();
+                goto case;
+
             case TOK.semicolon:
```

Comment:
This looks like it unintentionally bleeds into D code parser. Did you forget an `if (endOfLine) { ... }`?